_sql-and-ppl/ppl/commands/join.md

```diff
@@ -0,0 +1,218 @@
+---
+layout: default
+title: join
+parent: Commands
+grand_parent: PPL
+nav_order: 21
+---
+
+# join
+
+The `join` command combines two datasets. The left side can be an index or the results of piped commands, while the right side can be either an index or a subsearch.
+
+## Syntax
+
+The `join` command supports basic and extended syntax options.
+
+### Basic syntax
+
+```sql
+[joinType] join [left = <leftAlias>] [right = <rightAlias>] (on | where) <joinCriteria> <right-dataset>
+```
+
+When using aliases, `left` must appear before `right`.
+{: .note}
+
+The following are examples of the basic `join` command syntax:
+
+```sql
+source = table1 | inner join left = l right = r on l.a = r.a table2 | fields l.a, r.a, b, c
+source = table1 | inner join left = l right = r where l.a = r.a table2 | fields l.a, r.a, b, c
+source = table1 | left join left = l right = r on l.a = r.a table2 | fields l.a, r.a, b, c
+source = table1 | right join left = l right = r on l.a = r.a table2 | fields l.a, r.a, b, c
+source = table1 | full left = l right = r on l.a = r.a table2 | fields l.a, r.a, b, c
+source = table1 | cross join left = l right = r on 1=1 table2
+source = table1 | left semi join left = l right = r on l.a = r.a table2
+source = table1 | left anti join left = l right = r on l.a = r.a table2
+source = table1 | join left = l right = r [ source = table2 | where d > 10 | head 5 ]
+source = table1 | inner join on table1.a = table2.a table2 | fields table1.a, table2.a, table1.b, table1.c
+source = table1 | inner join on a = c table2 | fields a, b, c, d
+source = table1 as t1 | join left = l right = r on l.a = r.a table2 as t2 | fields l.a, r.a
+source = table1 as t1 | join left = l right = r on l.a = r.a table2 as t2 | fields t1.a, t2.a
+source = table1 | join left = l right = r on l.a = r.a [ source = table2 ] as s | fields l.a, s.a
+```
+
+#### Basic syntax parameters
+
+The basic `join` syntax supports the following parameters.
+
+| Parameter | Required/Optional | Description |
+| --- | --- | --- |
+| `<joinCriteria>` | Required | A comparison expression specifying how to join the datasets. Must be placed after the `on` or `where` keyword in the query. |
+| `<right-dataset>` | Required | The right dataset, which can be an index or a subsearch, with or without an alias. |
+| `joinType` | Optional | The type of join to perform. Valid values are `left`, `semi`, `anti`, and performance-sensitive types (`right`, `full`, and `cross`). Default is `inner`. |
+| `left` | Optional | An alias for the left dataset (typically a subsearch), used to avoid ambiguous field names. Specify as `left = <leftAlias>`. |
```

Comment:
```suggestion
| `left` | Optional | An alias for the left dataset (typically a subsearch) used to avoid ambiguous field names. Specify as `left = <leftAlias>`. |
```